WordPress/gutenberg
# [Pattern Directory]: Allow pattern registration from directory with theme.json

Resolves: https://github.com/WordPress/gutenberg/issues/35364
Depends on https://github.com/WordPress/gutenberg/pull/38625

This PR enables themes to register patterns from Pattern Directory through `theme.json`. It checks for a `patterns` top level key that is an array of `pattern slugs` from the directory. 


## Testing instructions

1. Merge https://github.com/WordPress/gutenberg/pull/38625 in this branch. `git merge --squash update/extend-theme-json`.
2. Use `empty` theme and go to the patterns tab in the inserter.
3. You can see that the two provided patterns from theme.json are registered([pattern1](https://wordpress.org/patterns/pattern/short-text-surrounded-by-round-images/), [pattern2](https://wordpress.org/patterns/pattern/partner-logos/)).

Alternatively, you can add in your theme.json some patterns in the `patterns` property.

## Notes
- You can see from the `second pattern` that is under `uncategorized`. That is because there is no parity between pattern directory categories and core categories(specifically `images` category in Pattern Directory.
